packages/graphile-build/src/SchemaBuilder.d.ts

```diff
@@ -0,0 +1,221 @@
+import {
+  GraphQLSchema,
+  GraphQLObjectType,
+  GraphQLInterfaceType,
+  GraphQLInputObjectType,
+  GraphQLEnumType,
+  GraphQLOutputType,
+  GraphQLFieldResolver,
+  GraphQLFieldConfigArgumentMap,
+  GraphQLFieldConfigMap,
+  GraphQLFieldConfig,
+  GraphQLEnumValueConfigMap,
+  GraphQLEnumValueConfig,
+  GraphQLInputFieldConfigMap,
+  GraphQLInputFieldConfig
+} from "graphql";
+import * as EventEmitter from "events";
+
+type mixed = {} | string | number | boolean | undefined | null;
+
+export interface Options {
+  [str: string]: mixed;
+}
+
+export type Plugin = (
+  builder: SchemaBuilder,
+  options: Options
+) => Promise<void> | void;
+
+type TriggerChangeType = () => void;
+export type WatchUnwatch = (triggerChange: TriggerChangeType) => void;
+
+export type SchemaListener = (newSchema: GraphQLSchema) => void;
+export type DataForType = {
+  [str: string]: Array<mixed>;
+};
+
+export type InitObject = {};
+
+export interface Build {
+  [str: string]: any;
+}
+export interface Inflection {
+  [str: string]: any;
+}
+export interface Scope<Type> {
+  [str: string]: any;
+}
+export interface Context<Type> {
+  scope: Scope<Type>;
+  [str: string]: any;
+}
```

Comment:
A number of `any`s here; when we finish moving to TypeScript these will end up being a lot more precise.

I ultimately plan to use [declaration merging](https://www.typescriptlang.org/docs/handbook/declaration-merging.html) so that each plugin can add its own types to these interfaces. This will make the plugin system a LOT easier to use 👍

Reply:
Perhaps, but I want people to be able to use it with TypeScript now rather than in a couple months when I finish moving it to TS. Mostly it was just copy and paste from Flow and then fix the syntax in a few places with Vim though 👍